pyo3-build-config/src/impl_.rs

```diff
@@ -484,6 +484,20 @@ fn search_lib_dir(path: impl AsRef<Path>, cross: &CrossCompileConfig) -> Vec<Pat
         };
         sysconfig_paths.extend(sysc);
     }
+    // If we got more than one file, only take those that contain the arch name.
+    // For ubuntu 20.04 with host architecture x86_64 and a foreign architecture of armhf
+    // this reduces the number of candidates to 1:
+    //
+    // $ find /usr/lib/python3.8/ -name '_sysconfigdata*.py' -not -lname '*'
+    //  /usr/lib/python3.8/_sysconfigdata__x86_64-linux-gnu.py
+    //  /usr/lib/python3.8/_sysconfigdata__arm-linux-gnueabihf.py
+    if sysconfig_paths.len() > 1 {
+        sysconfig_paths = sysconfig_paths
+            .iter()
+            .filter(|p| p.to_string_lossy().contains(&cross.arch))
+            .map(|p| p.clone())
+            .collect::<Vec<PathBuf>>();
+    }
```

Comment:
The only potential issue I see is that if the `arch` doesn't match _any_ sysconfig file (e.g. maybe the arch is spelt differently for some reason?) then this could reduce the number of candidates to zero.

In that case, it might be worth keeping the original full set of paths instead of listing zero paths and failing?